core/ui/compose/designsystem/src/main/kotlin/app/k9mail/core/ui/compose/designsystem/organism/drawer/ModalNavigationDrawer.kt

```diff
@@ -0,0 +1,48 @@
+package app.k9mail.core.ui.compose.designsystem.organism.drawer
+
+import androidx.compose.material3.DrawerValue
+import androidx.compose.material3.rememberDrawerState
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.rememberCoroutineScope
+import androidx.compose.ui.Modifier
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+import androidx.compose.material3.ModalNavigationDrawer as Material3ModalNavigationDrawer
+
+@Composable
+fun ModalNavigationDrawer(
+    drawerContent: @Composable (closeDrawer: () -> Unit) -> Unit,
+    modifier: Modifier = Modifier,
+    gesturesEnabled: Boolean = true,
+    content: @Composable (openDrawer: () -> Unit) -> Unit,
+) {
+    val drawerState = rememberDrawerState(initialValue = DrawerValue.Closed)
+    val scope = rememberCoroutineScope()
+
+    Material3ModalNavigationDrawer(
+        drawerContent = {
+            drawerContent {
```

Comment:
I found this hard to read. `drawerContent { … }` makes it look like it's a Composable container and the lambda sets its content.

The following refactoring would make it easier to read:

```kotlin
    // Alternatively, these could be inner functions
    val openDrawer: () -> Unit = {
        scope.launch {
            drawerState.open()
        }
    }

    val closeDrawer: () -> Unit = {
        scope.launch {
            delay(DRAWER_CLOSE_DELAY)
            drawerState.close()
        }
    }

    Material3ModalNavigationDrawer(
        drawerContent = {
            drawerContent(closeDrawer)
        },
        modifier = modifier,
        drawerState = drawerState,
        gesturesEnabled = gesturesEnabled,
        content = {
            content(openDrawer)
        },
    )
```

Reply:
Looks better